server/models/User.js

```diff
@@ -28,24 +30,43 @@ const userSchema = new mongoose.Schema({
 		required: true,
 		trim: true,
 		minLength: 6
-	}
+	},
+	// Lets users sign in on multiple devices w/ invalidating other tokens
+	tokens: [
+		{
+			token: {
+				type: String,
+				required: true
+			}
+		}
+	]
 });
 
+// Generating an auth token for our user
+userSchema.methods.generateAuthToken = async function () {
+	const user = this;
+	const token = jwt.sign({ _id: user._id.toString() }, "myToken");
+
+	// Concatenating and saving it to our document
+	user.tokens = user.tokens.concat({ token: token });
```

Comment:
hm, since you're signing your token by `jwt.sign({ _id: user._id.toString() }, "myToken");`, my thinking behind this is that it should be the _same_ token every single time so by concating it to `tokens`, you're concatenating the same token every time. can you verify this?

Reply:
Yeah, every time I sign in I get a new token and it gets concatenated. I forgot to mention it in the description but I address this in branch "5-file-upload-and-user-routes". In the new branch, every time I login it will use that token that we gave to the user so extra tokens do not get concatenated unless we sign in from a different device.

Reply:
oh okay, you probably don't need to store the token into your db because you can "calculate" (sign) it. The token will be passed in through cookies, or authorization header and stored on the browser level . Maybe i'll take a look at your new branch to see how you identify each device

Reply:
Ah, okay I'll keep that in mind when moving forward!

Reply:
Disregard what I said, I thought I addressed it but I am in fact concatenating the same token every time. Will work on this asap!